File: python/sorts/bubble_sort.py
def bubble_sort(arr: list):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]


def optimized_bubble_sort(arr: list):
    n = len(arr)
    for i in range(n - 1):
        swapped = False
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                if not swapped:
                    swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
        if not swapped:
            break

File: javascript/sorts/bubble_sort.js
require("../utils/array_utils/swap");
/** Bubble Sort Explanation
 * @param {Array} arr - The root array
 */
const bubbleSort = function BubbleSort(arr) {
	for (let i = 0; i < arr.length; i++) {
		for (let j = i + 1; j < arr.length - 1; j++) {
			if (arr[i] >= arr[j]) {
				arr.swap(arr, i, j);
			}
		}
	}
	return arr;
};

console.log(bubbleSort([2, 3, 1, 5, 2, 7]));
